### PR TITLE
`cloudflare_worker_domain` resource

### LINKED ISSUE #1921: Support Worker Domains
### Current Terraform and Cloudflare provider version

Latest

### Description

Support Worker Domains

Feature: https://developers.cloudflare.com/workers/platform/routing/custom-domains/
API: https://api.cloudflare.com/#worker-domain-properties

### Use cases

I would like to attach a domain to a worker

### Potential Terraform configuration

```hcl
...
```

### References

_No response_

### PR DESCRIPTION
**Pull request by abiodunakande**
Adds cloudflare_worker_domain resource following up on this [issue](https://github.com/cloudflare/terraform-provider-cloudflare/issues/1921)

Pending documentation

testpass

Closes #1921

### CODE REVIEW

**Comment by abiodunakande** [comment]
Struggling to run `make docs` on my M1 at the moment, would appreciate if someone could do the docs for me

File: internal/sdkv2provider/resource_cloudflare_workers_domain_test.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+)
+
+
+const (
+	scriptContent = `addEventListener('fetch', event => {event.respondWith(new Response('test'))});`
+)
+func TestAccCloudflareWorkerDomain_Attach(t *testing.T) {
+	var domain cloudflare.WorkersDomain
+	rnd := generateRandomResourceName()
+	hostname := rnd + ".com"
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_workers_domain_test.go:21]: we have a domain you can use via env vars `CLOUDFLARE_DOMAIN` that you can prefix a subdomain with a random set of characters

File: internal/sdkv2provider/resource_cloudflare_workers_domain_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+		}
+
+		*domain = foundDomain
+		return nil
+	}
+}
+
+func testAccCheckCloudflareWorkerDomainAttach(rnd, accountID string, hostname string) string {
+	return fmt.Sprintf(`
+resource "cloudflare_zone" "%[1]s-zone" {
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_workers_domain_test.go:91]: why are we creating a new zone here? we have a zone already that is setup and entitled correctly for Workers so we can use that (`CLOUDFLARE_ZONE_ID` and `CLOUDFLARE_ACCOUNT_ID` respectively)

File: internal/sdkv2provider/resource_cloudflare_workers_domain_test.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+resource "cloudflare_worker_script" "%[1]s-script" {
+  account_id = "%[3]s"
+  name = "%[1]s"
+  content = "%[2]s"
+}
+
+resource "cloudflare_worker_domain" "%[1]s" {
+	zone_id = "${cloudflare_zone.%[1]s-zone.id}"
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_workers_domain_test.go:103]: i think this should just be the existing zone we have here

**Comment by jacobbednarz** [comment]
> Struggling to run make docs on my M1 at the moment, would appreciate if someone could do the docs for me

you'll need to update your PR to allow maintainers to edit. it's checked by default when you raise a PR so it's something you explictly toggle off when opening a PR.

**Pull request closed without merging**